ifrs17/Report/ReportStorage.ipynb

```diff
@@ -221,6 +224,11 @@
                 "\n        var descendants = hierarchicalDimensionCache.Get<T>(systemName).Descendants(includeSelf: true);",
                 "\n        return descendants.Where(x => !descendants.Select(y => y.Parent).Contains(x.SystemName)).Select(x => x.SystemName).ToHashSet();",
                 "\n    }",
+                "\n",
+                "\n    public async Task<HashSet<string>> GetScenariosAsync(string scenario) => ",
+                "\n        scenario == Scenarios.Delta || scenario == Scenarios.All ",
+                "\n        ? (await workspace.Query<PartitionByReportingNodeAndPeriod>().Select(x => x.Scenario).ToArrayAsync()).ToHashSet()",
```

Comment:
As said, I think it's nicer (separation of concerns) to have 2 controls for this.

Reply:
It can also be done later.